src/TTextEdit.cpp

```diff
@@ -1784,6 +1805,433 @@ int TTextEdit::getRowCount()
+                                                    "<tr>%1</tr>"
+                                                    "<tr>%2</tr>"
+                                                    "<tr>%3</tr>"
+                                                    "<tr>%4</tr>"
+                                                    "<tr>%5</tr>"
+                                                    "<tr>%6</tr>"
+                                                    "</table></small><br>")
+                                     .arg(indexes, vals, elements, utf8Indexes, utf8Vals, luaCodes));
+            }
+            rowItems = 0;
+            indexes.clear();
+            vals.clear();
+            elements.clear();
+            utf8Indexes.clear();
+            utf8Vals.clear();
+            luaCodes.clear();
+        }
+    }
+
+    if (mpContextMenuAnalyser) {
+        if (isFirstRow) {
+            // if this is still true then we only have a short, single line of
+            // less than 16 codepoints
+            mpContextMenuAnalyser->setToolTip(QStringLiteral("<html><head/><body>%1"
+                                                             "<small><table border=\"1\" style=\"margin-top:5px; margin-bottom:5px; margin-left:5px; margin-right:5px;\" width=\"100%\" cellspacing=\"2\" cellpadding=\"0\">"
+                                                             "<tr><th>%2</th>%3</tr>"
+                                                             "<tr><th>%4</th>%5</tr>"
+                                                             "<tr><th>%6</th>%7</tr>"
+                                                             "<tr><th>%8</th>%9</tr>"
+                                                             "<tr><th>%10</th>%11</tr>"
+                                                             "<tr><th>%12</th>%13</tr>"
+                                                             "</table></small></body></html>")
+                                              .arg(completedRows)
+                                              .arg(tr("Index (UTF-16)", "1st Row heading for Text analyser output, table item is the count into the QChars/TChars that make up the text {this translation used 2 times}"), indexes)
+                                              .arg(tr("U+<i>####</i> Unicode Code-point <i>(High:Low Surrogates)</i>", "2nd Row heading for Text analyser output, table item is the unicode code point (will be between 000001 and 10FFFF in hexadecimal) {this translation used 2 times}"), vals)
+                                              .arg(tr("Visual", "3rd Row heading for Text analyser output, table item is a visual representation of the character/part of the character or a '{'...'}' wrapped letter code if the character is whitespace or otherwise unshowable {this translation used 2 times}"), elements)
```

Comment:
I think visual should be the first row, as it's most important because it's something the user can identify with quickly.

Reply:
It is important but It forms the division between what the user sees (apart from combining diacriticals because they must get *uncombined* in this display) and the two separate representations as UTF-16 _above_ (what the user will need/encounter if entering things into the Editor trigger items) and UTF-8 _below_ (what the user will need if entering things into a lua script)...